doc/kdump/kdump_Remote_SSH_HLD.md

```diff
@@ -0,0 +1,226 @@
+
+```
+admin@sonic: ssh-copy-id username@server_address
+```
+
+ This helps automate passwordless SSH logins by copying public key to authorized servers.
+
+## kdump Remote Architechture
+![alt text](./images/kdump-ssh.drawio.png)
+
+## Functional Description
+
+### Design Changes
+
+The SONiC kernel core dump remote functionality can be divided into two categories:
+
+1. Kernel core-dump generation service
+2. Storing Kernel core-dump files remotely
+
+Current SONiC lacks remote kernel dump functionality. To add this feature, consider enabling kdump for remote storage.
+
+We are suggesting modifying the exisitng SONiC configuration as following sonic-buildimage files.
+ 
+1. build_debian.sh (Addition)
+
+    - Required for kdump_remote_ssh_dump: Initialize network interfaces and enable DHCP upon kernel crash. 
+
+2. files/scripts/network_setup.sh (New Addition)
+    - A script to initialize the network interfaces and enable DHCP on them.
+3. files/script/network_setup.hook (New Addition)
+
+## Configuration and Management
+This section describes all types of configuration and management related design. Example sub-sections for "CLI" and "Config DB" are given below.
+
+
+
+### CONFIG_DB Enhancements
+ New attributes will be introduced to "KDUMP" table in ConfigDB for maintaining remote kdump configurations. Below is the schema for this table.
+
+```
+    KDUMP_TABLE:{{config}}
+     "enabled"    :{{"false"|"true"}}
+     "memory"     :{{string}}
+     "num_dumps"  :{{number}}
+     "remote"     :{{"false"|"true"}}
+     "ssh_connection_string" :{{string}}
+     "ssh_private_key_path" :{{string}}
+```
+
+### SAI API 
+
+No SAI API change or addition is needed for this HLD. 
+
+
+### CLI/YANG model Enhancements 
+### CLI Enhancements
+
+#### CLI Configuration Commands
+New SONiC CLI commands are introduced to configure remote kdump feature.
+
+```
+admin@sonic: sudo config kdump remote enable 
+```
+This commans will configure remote kdump feature in SONiC.
+
+```
+admin@sonic: sudo config kdump remote ssh_connection_string
+```
+
+```
+admin@sonic: sudo config kdump remote ssh_private_key_path
+```
+
+```
+admin@sonic: sudo config kdump remote disable
+```
+This command disables the remote kdump feature.
+#### CLI Show Commands
+An existing SONiC CLI command is used to display the current remote kdump feature configuraitons.
+
+```
+admin@sonic: show kdump config
+```
+
+Example output from the above command:
+```
+admin@sonic:~$ show kdump config
+Kdump administrative mode: Enabled
+Kdump operational mode: Ready
+Kdump memory reservation: 512
+Maximum number of Kdump files: 3
+remote: true
+Kdump remote server user@ip/hostname: ghulam@172.16.7.13
+Kdump private key file path for remote ssh connection: /home/admin/.ssh/id_rsa
+```
+
+### YANG Enhancements
+
+```
+leaf remote {
+   description "SSH Remote Config";
+
+					type string {
+						pattern "true|false";
+					} 
+				}
+
+leaf ssh_connection_string {
```

Comment:
Is it possible to add more than one remote IPs? e.g if one of the server is not reachable, is it possible to choose the alternate server to transfer the kdump data?

Reply:
No this is not possible according to my understanding. There is variable #SSH in kdump-config file in /etc/default/kdump-tools directory, it allows you to add only one remote IP at a time.